packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -505,25 +474,31 @@ function parseIf(
         ctx.throwOnNode(ParserDiagnostics.UNEXPECTED_IF_MODIFIER, ifAttribute, [modifier]);
     }
 
-    return ast.ifNode(
+    const node = ast.ifNode(
         modifier,
         ifAttribute.value,
         ast.sourceLocation(parse5ElmLocation),
         ifAttribute.location
     );
+
+    ctx.addNodeCurrentScope(node);
+    parent.children.push(node);
+
+    return node;
 }
 
 function parseIfBlock(
     ctx: ParserCtx,
     parsedAttr: ParsedAttribute,
-    parse5ElmLocation: parse5.ElementLocation
+    parse5ElmLocation: parse5.ElementLocation,
+    parent: ParentNode
 ): IfBlock | undefined {
     const ifBlockAttribute = parsedAttr.pick('lwc:if');
     if (!ifBlockAttribute) {
         return;
     }
 
-    const hasElseBlock = ctx.findSibling(ast.isIfBlock);
+    const hasElseBlock = ctx.findSibling(ast.isElseBlock);
```

Comment:
I think we should rename `findSibling` in the `ctx` since there's a new sibling array, maybe we can call it something like `findInCurrentScope`.